pkg/cloudevents/extensions.go

```diff
@@ -27,4 +27,4 @@ func caseInsensitiveSearch(key string, space map[string]interface{}) (interface{
 	return nil, false
 }
 
-var IsAlphaNumericLowercaseLetters = regexp.MustCompile(`^[a-z0-9]+$`).MatchString
+var IsAlphaNumeric = regexp.MustCompile(`^[a-zA-Z0-9]+$`).MatchString
```

Comment:
the spec says only lowercase are supported.

Reply:
```markdown
CloudEvents attribute names MUST consist of lower-case letters ('a' to 'z') or digits ('0' to '9') from the ASCII character set. Attribute names SHOULD be descriptive and terse and SHOULD NOT exceed 20 characters in length.
```

https://github.com/cloudevents/spec/blob/master/spec.md#attribute-naming-convention

Reply:
Yes but `setExtension` make it lowercase https://github.com/cloudevents/sdk-go/blob/2676232a051b22008883dc9b95906842f017398b/pkg/cloudevents/eventcontext_v1.go#L82

Reply:
And this also makes the behaviour consistent with other context implementations where this is not required